### PR TITLE
Improve Modal a11y

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

A few changes to improve our `<Modal/>` component's accessibility. Needed a library, to handle focus locking when a modal is open, and turns out it (`react-focus-on`) handles a few of the things we were already doing, or needed to anyways. 

I also added a `renderWithContext()` helper for component tests that wraps `render()` from `@testing-library/react` for cases where some of our app's contexts are needed. I just added the react-query context for now, but we can add other necessary contexts for tests that need them as we bump into them.

* use `<Portal/>` to render it as a direct child of the body (not as a descendant of the app dom root).
* add various `aria-*` attributes to help screen readers
* Lock focus control to modal when it's visible
* Focus first tabbable element in the body automatically
* Prevent scrolling on `document.body` when modal is open so background doesn't scroll
* Make the `X` close icon a button so it's tabbable/focusable
* Clicking off the modal will close it now

## Related Issue(s)

Fixes WEB-393

## How to test

* Navigate around the Dashboard and try the various places modals are shown. Try tabbing through them.
* Try both desktop sized, and smaller screens like mobile

I ran through a test of the modals on the following browsers just to ensure there's no regressions.
- [x] Safari desktop
- [x] Safari iOS
- [x] Chrome (Mac)
- [x] Firefox (Mac)
- [x] Arc (Mac)
- [x] Brave (Mac)
- [x] Edge on Windows

The only thing noteworthy was a bug with Safari (iOS and desktop) where the footer buttons don't make it into the tab rotation for some reason. I'll look into this in a followup. The current state of tabbing in modals doesn't support footer buttons either, so fwiw this is no worse in that case.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-web-3e7b20ecc3a</li>
	<li><b>🔗 URL</b> - <a href="https://brad-web-3e7b20ecc3a.preview.gitpod-dev.com/workspaces" target="_blank">brad-web-3e7b20ecc3a.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

**Review by easyCZ — APPROVED** [review]
LGTM, but you may want eyes from someone more fluent in React

File: components/dashboard/src/components/Modal.tsx [hunk 1]
```diff
@@ -26,97 +28,128 @@ export default function Modal(props: {
...
-                >
-                    {props.closeable !== false && (
+        <Portal>
+            {/* backdrop overlay */}
+            <div
+                className="fixed top-0 left-0 bg-black bg-opacity-70 z-50 w-screen h-screen focus:ring-0"
+                onKeyDown={handleKeydown}
+                tabIndex={0}
+            >
+                {/* Modal outer-container for positioning */}
```
Comment (AlexTugarev) [inline components/dashboard/src/components/Modal.tsx:105]: 🥇 for commenting on this level of details!

**Review by AlexTugarev — APPROVED** [review]
Changes looks great and it works nicely.

**Comment by AlexTugarev** [comment]
> The only thing noteworthy was a bug with Safari (iOS and desktop) where the footer buttons don't make it into the tab rotation for some reason.

This is a strange bug indeed. First I thought it would be related the Tab Bar in iOS, but after switching to the single tab option in Settings > Safari, still the reduced navigation bar, which is half the height, cuts the viewport and doesn't let you use the footer buttons 🤷🏻‍♂️

**Review by gtsiolis — APPROVED** [review]
Thanks for making these changes, @selfcontained. 

- use `<Portal/>` to render it as a direct child of the body (not as a descendant of the app dom root).
🙌  — Don't know what this is but it sounds cool. 

- add various `aria-*` attributes to help screen readers
🧙 

- Lock focus control to modal when it's visible
💯 — Had some fun with pressing and holding tab in modals a couple of times.

- Focus first tabbable element in the body automatically
✨ — Thanks for also keeping the cancel action as the default focused element on confirmation modals for deleting items. Focus handling could be different based on the modal type, but this is already ten times better than what we have now. Maturing our modals with disabled buttons or migrating them to full pages could be a next step.

- Prevent scrolling on `document.body` when modal is open so background doesn't scroll
💯 — Yas!

- Make the `X` close icon a button so it's tabbable/focusable
🔮 — Woohoo!

- Clicking off the modal will close it now
👀 — This used to be the case for most modals but it's good to have this consistent with all cases.

**Pull request merged**